pkg/sql/compile/alter.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2023 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compile
+
+import (
+	"github.com/matrixorigin/matrixone/pkg/pb/api"
+	"github.com/matrixorigin/matrixone/pkg/pb/plan"
+	"github.com/matrixorigin/matrixone/pkg/vm/engine"
+)
+
+func (s *Scope) AlterTableCopy(c *Compile) error {
+	qry := s.Plan.GetDdl().GetAlterTable()
+	dbName := c.db
+	tblName := qry.GetTableDef().GetName()
+
+	// 1. lock origin table
+	if err := lockMoTable(c, dbName, tblName); err != nil {
```

Comment:
lock mo table and lock table must handle retry error.

Reply:
Okay, thank you. I have added the judgment on the lock table retry error and made modifications to other clerical errors, @zhangxu19830126